electrum/util.py

```diff
@@ -677,6 +678,15 @@ def format_satoshis(
     if len(fract_part) < num_zeros:
         fract_part += "0" * (num_zeros - len(fract_part))
     result = integer_part + DECIMAL_POINT + fract_part
+    # add whitespaces as thousands' separator for better readability of numbers
+    if add_thousands_sep:
+        integer_part = "{:,}".format(int(integer_part)).replace(',', ' ')
+        if len(fract_part) > 3:
+            fractional_part = fract_part
+            fract_part = ''
+            for i in range(0,len(fractional_part),3):
+                fract_part += fractional_part[i:i+3] + ' '
```

Comment:
I think this whole block can just be replaced like this :)
```suggestion
        fract_part = " ".join(fract_part[i:i+3] for i in range(0, len(fract_part), 3))
```